### PR TITLE
Configure correct MTU on AWS seed and shoot nodes

### PR DESCRIPTION
**Pull request by zanetworker**
**What this PR does / why we need it**:

The default MTU values configured for AWS instances are too high, this can lead to connectivity problems. This PR configures the MTU on both the seed and the shoot nodes to comply with the rest of the components. 

**Release note**:
```improvement operator
MTU values on AWS nodes are now changed to 1460 instead of 9000. 
```

### CODE REVIEW

File: controllers/provider-aws/charts/provider-aws/templates/daemonset-custom-mtu.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      k8s-app: mtu-customizer
+  template:
+    metadata:
+      labels:
+        k8s-app: mtu-customizer
+    spec:
+      hostNetwork: true
+      initContainers:
+      - name: mtu-customizer
+        image: alpine:3.10.0
```
Comment (rfranzke) [inline controllers/provider-aws/charts/provider-aws/templates/daemonset-custom-mtu.yaml]: This should not be hard-coded here but get injected out of the image vector.

File: controllers/provider-aws/charts/provider-aws/templates/daemonset-custom-mtu.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        command: ["ip", "link", "set", "dev", "eth0", "mtu", "1460"]
+        securityContext:
+          capabilities:
+            add: ["NET_ADMIN", "SYS_TIME"]
+      containers:
+        - name: pause
+          image: k8s.gcr.io/pause:3.1
```
Comment (rfranzke) [inline controllers/provider-aws/charts/provider-aws/templates/daemonset-custom-mtu.yaml]: Same here

File: controllers/provider-alicloud/pkg/webhook/controlplane/ensurer.go [hunk 3]
```diff
@@ -116,3 +118,13 @@ func (e *ensurer) EnsureKubeletConfiguration(ctx context.Context, kubeletConfig
 	kubeletConfig.FeatureGates["KubeletPluginsWatcher"] = true
 	return nil
 }
+
+// EnsureAdditionalUnits ensures that additional required system units are added.
+func (e *ensurer) EnsureAdditionalUnits(ctx context.Context, units *[]extensionsv1alpha1.Unit) error {
```
Comment (rfranzke) [inline controllers/provider-alicloud/pkg/webhook/controlplane/ensurer.go]: Instead of implementing `EnsureAdditionalUnits` and `EnsureAdditionalFiles` this for every individual `ensurer` please implement this here: https://github.com/gardener/gardener-extensions/blob/master/pkg/webhook/controlplane/genericmutator/noopensurer.go

Then you can save all this code and don't need to define the method at all for all the different webhook ensurers that don't want to implement it.

File: controllers/provider-aws/pkg/webhook/controlplane/ensurer.go [hunk 4]
```diff
@@ -205,3 +204,42 @@ func (e *ensurer) EnsureKubernetesGeneralConfiguration(ctx context.Context, data
...
+	for _, un := range *units {
+		if un.Name == unit.Name {
+			return
+		}
+	}
+	*units = append(*units, unit)
+}
+
+// EnsureAdditionalUnits ensures that additional required system units are added.
+func (e *ensurer) EnsureAdditionalUnits(ctx context.Context, units *[]extensionsv1alpha1.Unit) error {
```
Comment (rfranzke) [inline controllers/provider-aws/pkg/webhook/controlplane/ensurer.go]: Can you add a test for this in https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-aws/pkg/webhook/controlplane/ensurer_test.go?

File: controllers/provider-aws/pkg/webhook/controlplane/ensurer.go [hunk 5]
```diff
@@ -205,3 +204,42 @@ func (e *ensurer) EnsureKubernetesGeneralConfiguration(ctx context.Context, data
 	*data = buf.String()
 	return nil
 }
+
+func appendUniqueUnit(units *[]extensionsv1alpha1.Unit, unit extensionsv1alpha1.Unit) {
```
Comment (rfranzke) [inline controllers/provider-aws/pkg/webhook/controlplane/ensurer.go]: Does it make sense to move this to https://github.com/gardener/gardener-extensions/blob/master/pkg/webhook/controlplane/utils.go?

File: controllers/provider-aws/pkg/webhook/controlplane/ensurer.go [hunk 6]
```diff
@@ -205,3 +204,42 @@ func (e *ensurer) EnsureKubernetesGeneralConfiguration(ctx context.Context, data
...
+		Enable:  &trueVar,
+		Command: &command,
+		Content: &customMTUUnitContent,
+	})
+
+	return nil
+}
+
+// EnsureAdditionalFiles ensures that additional required system files are added.
+func (e *ensurer) EnsureAdditionalFiles(ctx context.Context, files *[]extensionsv1alpha1.File) error {
```
Comment (rfranzke) [inline controllers/provider-aws/pkg/webhook/controlplane/ensurer.go]: See above, when implementing this in the `NoopEnsurer` you wouldn't need to implement it here.

File: controllers/provider-aws/charts/provider-aws/templates/daemonset-custom-mtu.yaml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+{{ if eq .Values.gardener.seed.provider "aws" }}
+kind: DaemonSet
```
Comment (mvladev) [inline controllers/provider-aws/charts/provider-aws/templates/daemonset-custom-mtu.yaml]: It needs [tolerations](https://github.com/gardener/garden/blob/d7258fb99fbbc8f5a674c7459e04240a8ff094fd/charts/shoot-core/charts/kube-proxy/templates/kube-proxy-daemonset.yaml#L50-L56) and custom [priority class](https://github.com/gardener/gardener/blob/d7258fb99fbbc8f5a674c7459e04240a8ff094fd/charts/shoot-core/charts/kube-proxy/templates/kube-proxy-daemonset.yaml#L49) as `system-cluster-critical` can only be used in `kube-system` NS.

Comment (zanetworker) [reply controllers/provider-aws/charts/provider-aws/templates/daemonset-custom-mtu.yaml]: but I am not using `system-cluster-critical`, or do you suggest that it needs to be `system-cluster-critical` ?

Comment (zanetworker) [reply controllers/provider-aws/charts/provider-aws/templates/daemonset-custom-mtu.yaml]: @mvladev btw, after 1.11 you can use `critical` for any namespace: https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/#marking-pod-as-critical

File: .ci/component_descriptor [hunk 8]
```diff
@@ -23,4 +23,29 @@ for images_yaml_path in "$dir/../controllers/"*"/charts/images.yaml"; do
...
+  IFS=' ' read -r -a array <<< "$line"
+  IFS=': ' read -r -a imageAndTag <<< ${array[1]}
+
+  NAME=${array[0]}
+  REPOSITORY=${imageAndTag[0]}
+  TAG=${imageAndTag[1]}
+
+  gardener="eu.gcr.io/gardener-project/gardener"
+  if [[ "$NAME" == "hyperkube" ]]; then
+    ${ADD_DEPENDENCIES_CMD} --generic-dependencies "'{\"name\": \"$NAME\", \"version\": \"$TAG\"}'"
```
Comment (rfranzke) [inline .ci/component_descriptor]:
```suggestion
    ${ADD_DEPENDENCIES_CMD} --generic-dependencies "{\"name\": \"$NAME\", \"version\": \"$TAG\"}"
```

File: .ci/component_descriptor [hunk 9]
```diff
@@ -23,4 +23,29 @@ for images_yaml_path in "$dir/../controllers/"*"/charts/images.yaml"; do
...
+  elif [[ $REPOSITORY =~ "eu.gcr.io/gardener-project/gardener*" ]]; then
+    ${ADD_DEPENDENCIES_CMD} --generic-dependencies "'{\"name\": \"$NAME\", \"version\": \"$TAG\"}'"
```
Comment (rfranzke) [inline .ci/component_descriptor]:
```suggestion
    ${ADD_DEPENDENCIES_CMD} --generic-dependencies "{\"name\": \"$NAME\", \"version\": \"$TAG\"}"
```

File: .ci/component_descriptor [hunk 10]
```diff
@@ -23,4 +23,29 @@ for images_yaml_path in "$dir/../controllers/"*"/charts/images.yaml"; do
...
+  else
+    ${ADD_DEPENDENCIES_CMD} \"--container-image-dependencies "'{\"name\": \"${controller_dir}-${NAME}\", \"image_reference\": \"${REPOSITORY}:${TAG}\", \"version\": \"$TAG\"}'"
```
Comment (rfranzke) [inline .ci/component_descriptor]:
```suggestion
    ${ADD_DEPENDENCIES_CMD} --container-image-dependencies "{\"name\": \"${controller_dir}-${NAME}\", \"image_reference\": \"${REPOSITORY}:${TAG}\", \"version\": \"$TAG\"}"
```

File: controllers/provider-alicloud/pkg/webhook/controlplanebackup/ensurer.go [hunk 11]
```diff
@@ -22,7 +22,6 @@ import (
 	extensionscontroller "github.com/gardener/gardener-extensions/pkg/controller"
 	"github.com/gardener/gardener-extensions/pkg/webhook/controlplane"
 	"github.com/gardener/gardener-extensions/pkg/webhook/controlplane/genericmutator"
-
```
Comment (rfranzke) [inline controllers/provider-alicloud/pkg/webhook/controlplanebackup/ensurer.go]: Please revert all such unrelated changes in the import order.
